radixdlt/src/test/java/com/radixdlt/consensus/sync/SyncManagerTest.java

```diff
@@ -0,0 +1,232 @@
+package com.radixdlt.consensus.sync;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongSupplier;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.consensus.VertexMetadata;
+import com.radixdlt.middleware2.CommittedAtom;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SyncManagerTest {
+
+	private static CommittedAtom buildWithVersion(long version) {
```

Comment:
These tests need some way to manage the internal resources consumed by SyncManager.

For example, right now each SyncManager has an internal thread that it uses for processing.  One of these threads gets created for each SyncManager that is created here, consuming resources which are never released until the JVM exits.  These resources need to be cleaned up after use.

Suggest that a close() method is added to SyncManager that terminates the executor, and that close method is called in a "finally" block for each test.

Reply:
Ok.